File: android/src/main/java/com/reactnativechimesdk/EventEmitter.java
package com.reactnativechimesdk;

import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.bridge.WritableNativeMap;
import com.facebook.react.modules.core.DeviceEventManagerModule;
import com.reactnativechimesdk.data.RosterAttendee;

public class EventEmitter {

  private static final String MEETING_EVENT = "onChimeMeetingEvent";
  private static final String MEETING_STATE_CHANGE = "meetingStateChange";

  public static final String MEETING_USER_JOIN = "sinkMeetingUserJoin";
  public static final String MEETING_USER_LEFT = "sinkMeetingUserLeft";
  public static final String MEETING_ACTIVE_SHARE = "sinkMeetingActiveShare";
  public static final String MEETING_AUDIO_STATUS_CHANGE = "onSinkMeetingAudioStatusChange";
  public static final String MEETING_VIDEO_STATUS_CHANGE = "onSinkMeetingVideoStatusChange";

  public static final String KEY_EVENT = "event";
  public static final String KEY_DES = "des";
  public static final String KEY_USER_NAME = "userName";
  public static final String KEY_USER_ID = "userID";
  public static final String KEY_VIDEO_STATUS = "videoStatus";
  public static final String KEY_AUDIO_STATUS = "audioStatus";
  public static final String KEY_SHARE_STATUS = "shareStatus";

  public static final int SHARE_STATUS_START = 1;
  public static final int SHARE_STATUS_STOP = 4;

  public static void sendMeetingStateEvent(ReactContext context, String des) {
    WritableMap map = new WritableNativeMap();
    map.putString(KEY_EVENT, MEETING_STATE_CHANGE);
    map.putString(KEY_DES, des);
    sendEvent(context, map);
  }

  public static void sendMeetingUserEvent(ReactContext context, String event, RosterAttendee attendee, boolean isCameraOn) {
    if (attendee == null) {
      return;
    }
    WritableMap map = new WritableNativeMap();
    map.putString(KEY_EVENT, event);
    map.putString(KEY_USER_ID, attendee.getAttendeeId());
    map.putString(KEY_USER_NAME, attendee.getAttendeeName());
    map.putBoolean(KEY_AUDIO_STATUS, attendee.isMuted());
    map.putBoolean(KEY_VIDEO_STATUS, isCameraOn);
    sendEvent(context, map);
  }

  public static void sendMeetingUserShareEvent(ReactContext context, String event, RosterAttendee attendee, boolean isCameraOn, int shareStatus) {
    if (attendee == null) {
      return;
    }
    WritableMap map = new WritableNativeMap();
    map.putString(KEY_EVENT, event);
    map.putString(KEY_USER_ID, attendee.getAttendeeId());
    map.putString(KEY_USER_NAME, attendee.getAttendeeName());
    map.putBoolean(KEY_AUDIO_STATUS, attendee.isMuted());
    map.putBoolean(KEY_VIDEO_STATUS, isCameraOn);
    map.putInt(KEY_SHARE_STATUS, shareStatus);
    sendEvent(context, map);
  }

  private static void sendEvent(ReactContext context, WritableMap params) {
    context.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit(MEETING_EVENT, params);
  }
}

File: android/src/main/java/com/reactnativechimesdk/EventEmitter.kt
package com.reactnativechimesdk

import com.amazonaws.services.chime.sdk.meetings.audiovideo.AttendeeInfo
import com.facebook.react.bridge.ReactContext
import com.facebook.react.bridge.WritableMap
import com.facebook.react.bridge.WritableNativeMap
import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter
import com.reactnativechimesdk.data.RosterAttendee
import com.reactnativechimesdk.utils.getAttendeeName

class EventEmitter {

  companion object {
    private const val MEETING_EVENT = "onChimeMeetingEvent"
    private const val KEY_EVENT = "event"
    private const val KEY_DES = "des"
    var KEY_USER_NAME = "userName"
    var KEY_USER_ID = "userID"
    var KEY_VIDEO_STATUS = "videoStatus"
    var KEY_AUDIO_STATUS = "audioStatus"
    var KEY_VIDEO_RATIO = "videoRatio"
    var KEY_IS_HOST = "isHost"
    var KEY_SHARE_STATUS = "shareStatus"

    private const val MEETING_STATE_CHANGE = "meetingStateChange"
    const val MEETING_USER_JOIN = "sinkMeetingUserJoin"
    const val MEETING_USER_LEFT = "sinkMeetingUserLeft"
    const val MEETING_ACTIVE_SHARE = "sinkMeetingActiveShare"
    const val MEETING_AUDIO_STATUS_CHANGE = "onSinkMeetingAudioStatusChange"
    const val MEETING_VIDEO_STATUS_CHANGE = "onSinkMeetingVideoStatusChange"

    fun sendMeetingStateEvent(context: ReactContext, des: String) {
      val map: WritableMap = WritableNativeMap()
      map.putString(KEY_EVENT, MEETING_STATE_CHANGE)
      map.putString(KEY_DES, des)
      sendEvent(context, map)
    }

    fun sendMeetingUserEvent(context: ReactContext, event: String, attendee: RosterAttendee) {
      val map: WritableMap = WritableNativeMap()
      map.putString(KEY_EVENT, event)
      map.putString(KEY_USER_ID, attendee.attendeeId)
      map.putString(KEY_USER_NAME, attendee.attendeeName)
      map.putBoolean(KEY_AUDIO_STATUS, attendee.isMuted)
      // map.putBoolean(KEY_VIDEO_STATUS, attendee.isCameraOn)
      sendEvent(context, map)
    }

    private fun sendEvent(context: ReactContext, params: WritableMap?) {
      context.getJSModule(RCTDeviceEventEmitter::class.java)
        .emit(MEETING_EVENT, params)
    }
  }
}
